Source/Scene/Cesium3DTileset.js

```diff
@@ -515,37 +553,69 @@ define([
-                                child.refining = true;
-                                // Store the plane mask so that the child can optimize based on its parent's returned mask
-                                child.parentPlaneMask = planeMask;
-                                stack.push(child);
+                        if (outOfCore) {
+                            for (k = 0; (k < childrenLength) && hasAvailableRequests(tiles3D); ++k) {
+                                child = children[k];
+// TODO: we could spin a bit less CPU here and probably above by keeping separate lists for unloaded/ready children.
+                                if (child.isContentUnloaded()) {
+                                    requestContent(tiles3D, child, outOfCore);
+                                }
                             }
                         }
                     } else {
-                        // Tile does not meet SEE and it is refinable. Refine to children in front-to-back order.
+                        // Tile does not meet SEE and its children are loaded.  Refine to them in front-to-back order.
                         for (k = 0; k < childrenLength; ++k) {
                             child = children[k];
-                            child.refining = false;
                             // Store the plane mask so that the child can optimize based on its parent's returned mask
                             child.parentPlaneMask = planeMask;
                             stack.push(child);
                         }
+
+                        if (defined(t.descendantsWithContent)) {
+                            refiningTiles.push(t);
+                        }
                     }
                 }
             }
         }
+
+        checkRefiningTiles(refiningTiles, tiles3D, frameState);
+    }
+
+    function checkRefiningTiles(refiningTiles, tiles3D, frameState) {
+        // In the common case, a tile that uses replacement refinement is refinable once all its
+        // children are loaded. However if it has an empty child, refining to its children would
+        // show a visible gap. In this case, the empty child's children (or further descendants)
+        // would need to be selected before the original tile is refinable. It is hard to determine
+        // this easily during the traversal, so this fixes the situation retroactively.
+        var descendant;
+        var refiningTilesLength = refiningTiles.length;
+        for (var i = 0; i < refiningTilesLength; ++i) {
+            var j;
+            var refinable = true;
+            var refiningTile = refiningTiles[i];
+            var descendantsLength = refiningTile.descendantsWithContent.length;
+            for (j = 0; j < descendantsLength; ++j) {
+                descendant = refiningTile.descendantsWithContent[j];
+                if (!descendant.selected) {
+                    // TODO: also check that its visible
```

Comment:
Since a lot of the work is in https://github.com/AnalyticalGraphicsInc/cesium/pull/3449 this is just a reminder for that PR.